### ISSUE TITLE
Display popup with doc before select method etc.

### ISSUE THREAD

**Issue by PixelT** [ternjs/tern_for_sublime#107]
It will be nice to display hint before select the method / function / variable etc. from autocomplete list:
![preview](http://i.imgur.com/RLqvX3A.png)
This option can be combined with another idea which I request: #85

**Comment by marijnh** [comment]
I do not understand what you are trying to suggest. What does 'hint' refer to?

**Comment by easyfrog** [comment]
'hint' maybe the function's parameters.

**Comment by PixelT** [comment]
@marijnh, I was thinking about something like this:
![](http://i.imgur.com/7v7DExA.png)

**Comment by marijnh** [comment]
Yes, that'd be cool, and the information is available to do it. Anybody want to create an implementation?

**Comment by tansongyang** [comment]
For everyone's information, I tried investigating what it would take to implement this, and I think it requires the ability to detect what the currently selected autocomplete option is. I wasn't able to find anything in the API, and I've yet to receive a response on the [forum](https://forum.sublimetext.com/t/detect-current-selected-autocompletion-option/16079).

**Comment by easyfrog** [comment]
I use the 'TernJS' the simple one to this.
It can preivew the parameters of function.
![654](https://cloud.githubusercontent.com/assets/10724325/12709381/557ea7de-c8e6-11e5-8019-aebe937aad18.png)

**Comment by PixelT** [comment]
Visual Studio Code also have a nice solution for this (description display under the method):
![](http://i.imgur.com/EVqLBiC.png)

**Comment by tansongyang** [comment]
I'm interested in the idea of specifying the parameter names, but the only potential issue with that is that Sublime Text's autocomplete dropdown has a hardcoded width according to [the forum](https://forum.sublimetext.com/t/change-the-width-of-the-auto-complete-box/4034). Are people OK wiith that?

**Comment by PixelT** [comment]
Maybe this feature can be easiest to implement, by using external sublimetext popup plugin? ( https://github.com/facelessuser/sublime-markdown-popups )

**Comment by PixelT** [comment]
Topic up...
Maybe @facelessuser can help here?
